fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/idformat/RecordIdFormat.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * RecordIdFormat.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.idformat;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class RecordIdFormat {
+    public enum FormatElementType implements FormatElement {
+        NONE(0),
+        NULL(1),
+        INT32(5),
+        INT32_OR_NULL(5),
+        INT64(9),
+        INT64_OR_NULL(9),
+        UUID_AS_STRING(18),
+        STRING_16(50);
```

Comment:
Isn't this `18` for a string up to `16` single-byte characters?

Reply:
The Tuple encoding for non-ascii characters is using 3 characters, so the total comes to 48 bytes, plus 2 for the beginning and the end.
As a result, I've switched the String encoding to just use the `String.getBytes()` that uses UTF=16's 2 bytes per character, so the total comes to 32 bytes.

Reply:
Changed this to 27 to fit within 3 dimensions while still allowing some non-ASCII characters